### DISCUSSION TITLE
Tasks are ending early due to running out of memory

### DISCUSSION THREAD

**Discussion by aagronow-playon** [artilleryio/artillery#2449] [Artillery]
This is my cli

`artillery run-fargate --region us-east-1 --count 4 --launch-config '{"cpu": 4096, "memory": 12288}' script.yml`

I am trying to increase the memory for the tasks because they are running out of memory right before the load test finishes. Even though I am setting the memory in launch-config to 12gb, the task is still getting created with 8gb. 

![image](https://github.com/artilleryio/artillery/assets/151669027/969e56a4-17f5-451b-8799-055ed2271de7)

I have a workaround, (reduce arrivalRate and increase --count to have same effective throughput), but I would like to figure out why I cannot increase the memory for the task.

Here is the phase I am using:

>   phases:
>     - duration: 1200
>       arrivalRate: 10
>       rampTo: 30
>       name: "Ramp Up"
>     - duration: 1800
>       arrivalRate: 30
>       name: "Sustained Load"
>     - duration: 1200
>       arrivalRate: 30
>       rampTo: 10
>       name: "Cool Down"
>     - duration: 120
>       arrivalRate: 0
>       name: "Finish Up"

Additionally you can see from this screenshot that the memory is steadily increasing during the run, right up until the tasks die because it runs out. Any help or insight would be appreciated. Thanks!
![image](https://github.com/artilleryio/artillery/assets/151669027/54ceefe9-946b-4793-bf83-53e115aef7bd)

**Comment by hassy** [comment]
Hi @aagronow-playon 👋 Odd that the memory setting isn't being applied. Is this on the most recent version of Artillery? (you can check with `artillery -v`)

Are these Playwright-based tests? They tend to be memory-hungry, so reducing arrivalRate and increasing count is indeed the right way to go to scale out.

**Reply by aagronow-playon** [reply]
Thanks for responding @hassy, I am on artillery version 2.0.4. These are just artillery scripts written in javascript, not playwright library. I am using them to load test our application's endpoints.
